#! /bin/sh /usr/share/dpatch/dpatch-run
## 38_arm_xpcom_optim.dpatch by Mike Hommey <glandium@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patch from Antti P Miettinen to fix small optimization problem with
## DP: newer gcc's on arm. bz#322806

@DPATCH@

=== xpcom/reflect/xptcall/src/md/unix/xptcinvoke_arm.cpp
==================================================================
--- xulrunner/xpcom/reflect/xptcall/src/md/unix/xptcinvoke_arm.cpp	(revision 22)
+++ xulrunner/xpcom/reflect/xptcall/src/md/unix/xptcinvoke_arm.cpp	(local)
@@ -212,7 +212,7 @@
     "add	sp, sp, r4	\n\t"	/* restore stack pointer		*/
     "mov	%0, r0		\n\t"	/* the result...			*/
     : "=r" (result)
-    : "r" (&my_params)
+    : "r" (&my_params), "m" (my_params)
     : "r0", "r1", "r2", "r3", "r4", "ip", "lr", "sp"
     );
     
